### PR TITLE
Refactor Layer component

### PR DESCRIPTION
**Pull request by nolleto**
#### What does this PR do?

Refactor `Layer` component with Hooks and update the broken tests for this component.

#### Where should the reviewer start?

`src/js/components/Layer/Layer.js`

#### What testing has been done on this PR?

I've run the storybook and check the `Layer` behavior and also run `yarn test`

#### How should this be manually tested?

`yarn test`

#### Any background context you want to provide?

N/A

#### What are the relevant issues?

#3394

#### Screenshots (if appropriate)

N/A

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

No

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible

### CODE REVIEW

File: src/js/components/Layer/__tests__/Layer-test.js [hunk 1]
```diff
@@ -203,20 +203,20 @@ describe('Layer', () => {
...
-    ref.current.setState({ islayerContainerAvailable: false });
-    expect(queryByTestId(document, 'test-layer-container')).toBeNull();
-
-    ref.current.componentDidMount();
     expect(queryByTestId(document, 'test-layer-container')).toMatchSnapshot();
+    expect(container.firstChild).not.toBeNull();
+
+    unmount();
+
+    expect(container.firstChild).toBeNull();
```
Comment (nolleto) [inline src/js/components/Layer/__tests__/Layer-test.js]: Please, if you had a better way to test it just let me know 😄

File: src/js/components/Layer/Layer.js [hunk 2]
```diff
@@ -1,78 +1,82 @@
...
-  render() {
-    const { islayerContainerAvailable } = this.state;
-
-    return islayerContainerAvailable
-      ? createPortal(<LayerContainer {...this.props} />, this.layerContainer)
-      : null;
-  }
-}
+    };
+  }, [animate, animation]);
```
Comment (ericsoderberghp) [inline src/js/components/Layer/Layer.js]: I think this means we will animate the Layer out and change the focus back if the user changes the `animate` or `animation` properties. While I don't think those are likely to change, something feels a bit odd about that behavior. It seems like we shouldn't touch the focus or animation unless the Layer is being unmounted. Would it be possible to leave the dependency array empty?

Comment (nolleto) [reply src/js/components/Layer/Layer.js]: Sure, I will do it.

**Comment by ShimiSun** [comment]
hi buddy, so you mind taking a look at the tabIndex change on the test snapshot?

**Comment by nolleto** [comment]
Apparently, jest snapshot is adding `tabindex` in theses two `input`s here. 🤔 
`src/js/components/SkipLinks/__tests__/SkipLink-test.js`
```js
const { container } = render(
  <Grommet>
    <SkipLinks id="skip-links">
      <SkipLink id="main" label="Main Content" />
      <SkipLink id="footer" label="Footer" />
    </SkipLinks>
    <div>
      <SkipLinkTarget id="main" />
      Main Content
      <input type="text" value="main content" onChange={() => {}} /> // <--- HERE
    </div>
    <footer>
      <SkipLinkTarget id="footer" />
      <input type="text" value="footer" onChange={() => {}} /> // <--- HERE
    </footer>
  </Grommet>,
);
```

So, I've updated the snapshot.

**Comment by ShimiSun** [comment]
Thanks for contributing! I'll be closing this PR since refactoring Layer to use hooks was taken care of in other PR effort.

**Pull request closed without merging**